src/Fixer/Casing/NativeClassCasingFixer.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author siad007
+ */
+final class NativeClassCasingFixer extends AbstractFixer
+{
+    private static $nativeClassNames;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        if (null === self::$nativeClassNames) {
+            foreach (get_declared_classes() as $class) {
```

Comment:
Personally, I use this project as a library sometimes, so it's not always guaranteed that this will only target native classes.